### PR TITLE
Tpwdcee 5739 my favorite payment

### PR DESCRIPTION
**Pull request by rinnhofer**

### CODE REVIEW

File: wirecard-woocommerce-extension/assets/js/creditcard.js [hunk 1]
```diff
@@ -35,304 +35,347 @@
...
-	return jQuery.ajax(
-		{
-			type: "POST",
-			url: phpVars.vault_url,
-			data: request,
-			dataType: "json"
-		}
-	);
+function replaceClassForSelector( selector, needle, replacement ) {
+	selector.removeClass( needle ).addClass( replacement );
```
Comment (flavius) [inline wirecard-woocommerce-extension/assets/js/creditcard.js]: Use `$` for variable names holding jQuery objects. This makes it obvious that the variables have methods which you can call.

File: wirecard-woocommerce-extension/assets/js/creditcard.js [hunk 2]
```diff
@@ -35,304 +35,347 @@
...
+	VAULT_TABLE_ID: "vault-table",
+	NEW_CARD_CONTENT_AREA: "#wc_payment_method_wirecard_new_credit_card",
+	NEW_CARD_CONTENT_AREA_IFRAME: "#wc_payment_method_wirecard_new_credit_card iframe",
+	SEAMLESS_SUBMIT_BUTTON: "seamless-submit",
+	SEAMLESS_FORM_CONTAINER: "wc_payment_method_wirecard_creditcard_form",
+	NONCE_SELECTOR: "#wc_payment_method_wirecard_creditcard_response_form input[name='cc_nonce']",
+	MESSAGE_CONTAINER: "#wd-creditcard-messagecontainer",
+	WD_TOKEN_ID_PREFIX: "wd-token-",
+	
+};
```
Comment (flavius) [inline wirecard-woocommerce-extension/assets/js/creditcard.js]: Call `Object.freeze(obj);` on these 5 objects to make them read-only.

File: wirecard-woocommerce-extension/assets/js/creditcard.js [hunk 3]
```diff
@@ -343,75 +425,132 @@ function onFormSubmitted( response ) {
...
-	submitVault()
-		.then( handleSubmitResult )
-		.fail( logError );
+/**
+ * Get saved tokens and init view
+ *
+ * @since 3.1.0
+ */
+function initializeTokenList() {
+	let hasSavedTokens = document.getElementById( Constants.VAULT_TABLE_ID );
```
Comment (flavius) [inline wirecard-woocommerce-extension/assets/js/creditcard.js]: Why aren't we using jQuery here?

File: wirecard-woocommerce-extension/assets/js/creditcard.js [hunk 4]
```diff
@@ -343,75 +425,132 @@ function onFormSubmitted( response ) {
...
-/*
- * Integration code
+/**
+ * Initializes general event handlers for the interface
+ *
+ * @since 2.4.0
  */
+function initializeEventHandlers()
+{
+	let seamlessButtonSubmit = document.getElementById( Constants.SEAMLESS_SUBMIT_BUTTON );
```
Comment (flavius) [inline wirecard-woocommerce-extension/assets/js/creditcard.js]: Why aren't we using jQuery here?

File: wirecard-woocommerce-extension/assets/js/creditcard.js (hunk 3 above)
Comment (ozada) [reply wirecard-woocommerce-extension/assets/js/creditcard.js]: We should try avoiding usage of jQuery in future.

File: wirecard-woocommerce-extension/assets/js/creditcard.js (hunk 4 above)
Comment (ozada) [reply wirecard-woocommerce-extension/assets/js/creditcard.js]: We should try avoiding usage of jQuery in future.

**Comment by flavius** [comment]
Review passed.

**Pull request merged**